actionpack/lib/action_dispatch/http/mime_negotiation.rb

```diff
@@ -139,6 +141,33 @@ def negotiate_mime(order)
         order.include?(Mime::ALL) ? format : nil
       end
 
+      class VariantInquirer # :nodoc:
+        delegate :==, :each, :empty?, to: :variants
+
+        def initialize(variants = [])
+          @variants = variants
+        end
+
+        def any?(*candidates)
+          (variants & candidates).any?
+        end
+
+        def to_ary
+          variants
+        end
+
+        private
+          attr_reader :variants
```

Comment:
IMO, it shouldn't. It's intended for internal use only. Doesn't make sense to call `request.variant.variants`.